mrmustard/lab/representations/data/gaussian_data.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import numpy as np
+
+from typing import Optional, TYPE_CHECKING, Union
+
+from mrmustard.lab.representations.data.matvec_data import MatVecData
+from mrmustard.math import Math
+from mrmustard.typing import Batch, Matrix, Scalar, Tensor, Vector
+
+
+if TYPE_CHECKING:  # This is to avoid the circular import issue with GaussianData<>QPolyData
+    from mrmustard.lab.representations.data.qpoly_data import QPolyData
+
+math = Math()
+
+
+class GaussianData(MatVecData):
+    r"""Gaussian data for certain representation objects.
+
+    Gaussian data is made of covariance, mean vector and coefficient. Each of these has a batch
+    dimension, and the length of the batch dimension is the same for all three.
+    These are the parameters of a linear combination of Gaussians, which is Gaussian if there is
+    only one contribution for each.
+    Each contribution parametrizes the Gaussian function:
+    `coeffs * exp(-0.5*(x-mean)^T cov^-1 (x-mean))`.
+
+    Args:
+        cov (Optional[Batch[Matrix]]):      covariance matrices (real symmetric)
+        means (Optional[Batch[Vector]]):    mean vector of the state (real), note that the 
+                                            dimension must be even
+        coeffs (Optional[Batch[Scalar]]):   coefficients (complex)
+
+    Raises:
+        ValueError: if neither means nor covariance is defined
+    """
+
+    def __init__(
+        self,
+        cov: Optional[Batch[Matrix]] = None,
+        means: Optional[Batch[Vector]] = None,
+        coeffs: Optional[Batch[Scalar]] = None,
+    ) -> None:
+        
+        if cov is not None or means is not None:  # at least one is defined -or both-
+
+            if cov is None:
+                means =  np.array(means)
```

Comment:
we shouldn't use numpy's methods, or we lose differentiability. Perhaps you're looking for `math.astensor`